checks-superstaq/checks_superstaq/requirements.py

```diff
@@ -197,38 +182,49 @@ def _check_package_versions(
 
 
 @functools.lru_cache()
-def _get_latest_version(package: str) -> str:
+def _get_latest_version(package: str, silent: bool) -> str:
+    """Retrieve the latest version of a package."""
+    local_version = _get_local_version(package)
+    pypi_version = _get_pypi_version(package, silent)
+    if not local_version and not pypi_version:
+        raise ModuleNotFoundError(f"Package not installed or found on PyPI: {package}")
+    return max(pypi_version or "0.0.0", local_version or "0.0.0", key=packaging.version.parse)
+
+
+def _get_local_version(package: str) -> Optional[str]:
+    """Retrieve the local version of a package (if installed)."""
+    base_package = package.split("[")[0]  # remove options: package_name[options] --> package_name
+    try:
+        return importlib.metadata.version(base_package)
```

Comment:
is there a way to do this such that it reflects changes to `_version.py` in editable installs before we re-run `pip install`? (this would be the case if we just checked `<module>.__version__`, but the package metadata is only updated during installation)

Reply:
Ahh good point.  Let me try.

Reply:
Okay: I return `module.__version__` if found.

Reply:
nice thanks!